File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int opening = 0, closing = 0, len = s.length();
        for(int i=0;i<len;i++){
            if(s.charAt(i) ==')')
                if(opening > 0) opening--;
                else closing++;
            else opening++;
        }
        return opening+closing;
    }
}

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.cs
public class Solution {
    public int MinAddToMakeValid(string s) {
        int opening = 0, closing = 0;
        foreach(char ch in s.ToCharArray()){
            if(ch == ')'){
                if(opening > 0)
                    opening--;
                else closing++;
            }else{
                opening++;
            }
        }
        return opening+closing;
    }
}